### PR TITLE
Launch script

### LINKED ISSUE #107: Launch script argument butchering
## Bug Report

**Current Behavior**
Running the current go script won't properly work with string args, f.e. the payload search NF uses a `-s` argument which accepts a string, if such string is `"GET /8K"` then the NF will only get `GET` as an argument instead of `GET /8K`. Somehow the bash script butchers the arg.

**Expected behavior/code**
Clearly we don't want that, it should be passed properly.

**Steps to reproduce**
Reproduced easily with payload search NF

**Possible Solution**
This technically will solve it
```
exec sudo $BINARY $DPDK_ARGS -- $ONVM_ARGS -- "$@"
```
But that doesn't work for -- -- -- args, so we need a better solution

### PR DESCRIPTION
**Pull request by kevindweb**
From [Issue 107](https://github.com/sdnfv/openNetVM/issues/107), go script couldn't handle multi-word string arguments

## Summary:
Specifically in the `payload_scan` NF, running with the required `./go.sh 1 -d 1 -s "My String Here"` would not work with multiple words. While `./go.sh 1 -d 1 -s computer` or "computer" works well, bash strips the quotes with variable assignment, so "computer here" would become computer here. Thus the c `getopt` would not recognize the word `here`. 

**Usage:**

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  Resolves #107  
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                | 👍
| New functionality        |
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: None  

**TODO before merging :**
 - [ ] PR is ready for review

## Test Plan:
Run multiple NFs with tons of arguments to ensure the go script doesn't fail. Make sure to run with DPDK arguments as well, to check that errors are handled correctly still. Only adding DPDK without ONVM args should fail, from previous edits to go scripts. 

- Sanity checks, assigned to @koolzz  @dennisafa 
  - Run make in /onvm and /examples directories
  - Code style, assigned to @koolzz  @dennisafa 

- Run linter
  - Check everything style related
  - Code design, assigned to @koolzz  @dennisafa 

- Check for scripting errors
  - Check that code is reusable
  - Code is clean, functions are concise
  - Verify that edge cases properly handled

- Performance, assigned to @koolzz  @dennisafa 
  - Run Speed Tester NF, report performance.

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm do your thing 😉

**Comment by onvm** [comment]
> @onvm do your thing 😉 

## CI Message

Your results will arrive shortly

**Comment by kevindweb** [comment]
@koolzz Why doesn't ONVM say Linter Passed at the end if there's no errors?

File: examples/start_nf.sh [hunk 1]
```diff
@@ -48,7 +48,11 @@ fi
 
 # Check if -- is present, if so parse dpdk/onvm specific args
 dash_dash_cnt=0
+arg_cnt=0
 for i in "$@" ; do
+  if [[ dash_dash_cnt -le 1 ]] ; then
```
Comment (koolzz) [inline examples/start_nf.sh]: Could we make it `-lt 2` (seems to me a bit more intuitive)

File: examples/start_nf.sh [hunk 2]
```diff
@@ -48,7 +48,11 @@ fi
 
 # Check if -- is present, if so parse dpdk/onvm specific args
 dash_dash_cnt=0
+arg_cnt=0
```
Comment (koolzz) [inline examples/start_nf.sh]: Maybe something like non_nf_arg_cnt to be more specific?

File: examples/start_nf.sh [hunk 3]
```diff
@@ -58,20 +62,21 @@ done
 if [[ $dash_dash_cnt -ge 2 ]]; then
   DPDK_ARGS="$DPDK_BASE_ARGS $(echo " ""$@" | awk -F "--" '{print $1;}')"
   ONVM_ARGS="$(echo " ""$@" | awk -F "--" '{print $2;}')"
-  NF_ARGS="$(echo " ""$@" | awk -F "--" '{print $3;}')"
+
```
Comment (koolzz) [inline examples/start_nf.sh]: extra space

**Comment by koolzz** [comment]
This one is next @onvm

**Comment by onvm** [comment]
> This one is next @onvm 

## CI Message

Your results will arrive shortly

**Pull request merged**